pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -539,6 +561,10 @@ async def main():
             model_settings: Optional settings to use for this model's request.
             usage_limits: Optional limits on model request count or token usage.
             usage: Optional usage to start with, useful for resuming a conversation or agents used in tools.
+            metadata: Optional metadata to attach to this run. Accepts a dictionary or a callable taking
+                [`RunContext`][pydantic_ai.tools.RunContext]. The resolved dictionary is shallow merged into the
+                agent's metadata (or any [`Agent.override`][pydantic_ai.agent.Agent.override]) with run-level keys
```

Comment:
Existing `agent.override`s always cause the `agent.run` value to be ignored, fully overwriting the agent + agent run original. We should have the same behavior here